internal/disk/disk_test.go

```diff
@@ -971,3 +971,118 @@ func TestEnsureDirectorySizes(t *testing.T) {
 	}
 
 }
+
+func TestMinimumSizesWithRequiredSizes(t *testing.T) {
+	assert := assert.New(t)
+
+	// math/rand is good enough in this case
+	/* #nosec G404 */
+	rng := rand.New(rand.NewSource(13))
+	pt := testPartitionTables["plain"]
+
+	type testCase struct {
+		Blueprint        []blueprint.FilesystemCustomization
+		ExpectedMinSizes map[string]uint64
+	}
+
+	testCases := []testCase{
+		{ // specify small /usr -> / and /usr get default size
+			Blueprint: []blueprint.FilesystemCustomization{
+				{
+					Mountpoint: "/usr",
+					MinSize:    1 * MiB,
+				},
+			},
+			ExpectedMinSizes: map[string]uint64{
+				"/usr": 3 * GiB,
+				"/":    1 * GiB,
+			},
+		},
+		{ // specify small / and /usr -> / and /usr get default size
+			Blueprint: []blueprint.FilesystemCustomization{
+				{
+					Mountpoint: "/",
+					MinSize:    1 * MiB,
+				},
+				{
+					Mountpoint: "/usr",
+					MinSize:    1 * KiB,
+				},
+			},
+			ExpectedMinSizes: map[string]uint64{
+				"/usr": 3 * GiB,
+				"/":    1 * GiB,
+			},
+		},
+		{ // big /usr -> / gets default size
+			Blueprint: []blueprint.FilesystemCustomization{
+				{
+					Mountpoint: "/usr",
+					MinSize:    10 * GiB,
+				},
+			},
+			ExpectedMinSizes: map[string]uint64{
+				"/usr": 10 * GiB,
+				"/":    1 * GiB,
+			},
+		},
+		{
+			Blueprint: []blueprint.FilesystemCustomization{
+				{
+					Mountpoint: "/",
+					MinSize:    10 * GiB,
+				},
+				{
+					Mountpoint: "/home",
+					MinSize:    1 * MiB,
+				},
+			},
+			ExpectedMinSizes: map[string]uint64{
+				"/":     10 * GiB,
+				"/home": 1 * GiB,
+			},
+		},
+		{ // no separate /usr and no size for / -> / gets sum of default sizes for / and /usr
+			Blueprint: []blueprint.FilesystemCustomization{
+				{
+					Mountpoint: "/opt",
+					MinSize:    10 * GiB,
+				},
+			},
+			ExpectedMinSizes: map[string]uint64{
+				"/opt": 10 * GiB,
+				"/":    1 * GiB,
+			},
+		},
+	}
+
+	for idx, tc := range testCases {
+		{ // without LVM
+			mpt, err := NewPartitionTable(&pt, tc.Blueprint, uint64(3*GiB), false, map[string]uint64{"/": 1 * GiB, "/usr": 3 * GiB}, rng)
```

Comment:
So this is a bit confusing. Because of the `/usr` is not specified as a filesystem customization and ends up on the same partition as the root `/`, then the root partition will be 1 GiB, but `/usr` won't be. So I'm wondering it the code should use the biggest size for a mountpoint pass in the `RequiredSizes` if it won't end up on a dedicated partition. 🤔

Reply:
Alternatively, we should explain the behavior in this regard in a comment (which I hope I didn't miss).

Reply:
@supakeen could you please comment on this before we merge the PR?

Reply:
The current comment says: `the sum of their subdirectories plus their own sizes.` I feel that that is clear enough that having a 2 GiB /usr inside a 1 GiB / ends up with a 3 GiB total /.

Reply:
@supakeen the referenced comment looks good to me, but the confusing thing is the last test case in this test. Maybe I'm not reading it correctly 😇 

```go
		{ // no separate /usr and no size for / -> / gets sum of default sizes for / and /usr
			Blueprint: []blueprint.FilesystemCustomization{
				{
					Mountpoint: "/opt",
					MinSize:    10 * GiB,
				},
			},
			ExpectedMinSizes: map[string]uint64{
				"/opt": 10 * GiB,
				"/":    1 * GiB,  // <-- here I would expect 4 GiB ( 1 GiB for "/' + 3 GiB for "/usr") 
			},
		},

```

I'm considering

```go
map[string]uint64{"/": 1 * GiB, "/usr": 3 * GiB}
```

passed to `NewPartitionTable()` as `requiredSizes` in this test...

Reply:
Right, so, it is indeed a bit confusing. `/` does end up being 4 GiB in that testcase which is larger than the ExpectedMinSizes so the test passes. I'll up the expected minsize to 4 GiB there so it actually tests what it's supposed to test (1 GiB `/` + 3 GiB `/usr`).